cloud/scope/internet_gateway_reconciler.go

```diff
@@ -37,15 +41,15 @@ func (s *ClusterScope) ReconcileInternetGateway(ctx context.Context) error {
 		return err
 	}
 	if igw != nil {
-		s.OCIClusterAccessor.GetNetworkSpec().Vcn.InternetGatewayId = igw.Id
+		s.OCIClusterAccessor.GetNetworkSpec().Vcn.InternetGateway.Id = igw.Id
```

Comment:
It will be converted to v1beta2. So the e2e test ran with templates defined as v1beta1. Before the reconcile is called when the object is fetched into memory, it will be converted to current version.